### PR TITLE
CD-72889 Sand: Change to native New Relic client

### PR DESCRIPTION
**Pull request by gubenyas**
* [Main JIRA ticket](https://coupadev.atlassian.net/browse/CD-72889)

reviewers:
- [x] @johnwu96822

### CODE REVIEW

File: cmd/server/handler.go [hunk 1]
```diff
@@ -93,13 +93,29 @@ func RunHost(c *config.Config) func(cmd *cobra.Command, args []string) {
...
+	// create newrelic app here
 	if newRelicLicense != "" && newRelicApp != "" {
-		n.Use(negronigorelic.New(newRelicLicense, newRelicApp, verbose))
-		logrus.Info("New Relic enabled!")
+		newrelicConfig := newrelic.NewConfig(newRelicApp, newRelicLicense)
+		newrelicConfig.Logger = nrlogrus.StandardLogger()
+		app, err := newrelic.NewApplication(newrelicConfig)
+		if err == nil {
+			newRelicHandler := negroni.HandlerFunc(func(rw http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
+		    txn := app.StartTransaction(r.URL.EscapedPath(), rw, r)
```
Comment (johnwu96822) [inline cmd/server/handler.go]: Please fix the indentation on these lines

File: cmd/server/handler_test.go [hunk 2]
```diff
@@ -39,15 +39,15 @@ func TestMiddleware(t *testing.T) {
 	logrus.SetOutput(os.Stderr)
 }
 
-func testUseNewRelicMiddleware(t *testing.T) {
+func testUseNewRelicMiddlewareWithWrongValues(t *testing.T) {
```
Comment (johnwu96822) [inline cmd/server/handler_test.go]: This new test needs to be added to TestMiddleware

**Comment by gubenyas** [comment]
@johnwu96822 done, thanks

File: cmd/server/handler.go [hunk 3]
```diff
@@ -93,13 +93,29 @@ func RunHost(c *config.Config) func(cmd *cobra.Command, args []string) {
...
+	// create newrelic app here
 	if newRelicLicense != "" && newRelicApp != "" {
-		n.Use(negronigorelic.New(newRelicLicense, newRelicApp, verbose))
-		logrus.Info("New Relic enabled!")
+		newrelicConfig := newrelic.NewConfig(newRelicApp, newRelicLicense)
+		newrelicConfig.Logger = nrlogrus.StandardLogger()
+		app, err := newrelic.NewApplication(newrelicConfig)
+		if err == nil {
+			newRelicHandler := negroni.HandlerFunc(func(rw http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
+			txn := app.StartTransaction(r.URL.EscapedPath(), rw, r)
```
Comment (johnwu96822) [inline cmd/server/handler.go]: These need to be further indented as they are under the scope of the func

File: cmd/server/handler.go [hunk 4]
```diff
@@ -93,13 +93,29 @@ func RunHost(c *config.Config) func(cmd *cobra.Command, args []string) {
 	}
 }
 
-//Get NewRelic license and app name from environment variables
 func useNewRelicMiddleware(n *negroni.Negroni, verbose bool) {
```
Comment (johnwu96822) [inline cmd/server/handler.go]: Do we not need the verbose option any more? If so, this and the verbose command flag (in /cmd/host.go) can be taken out.

File: cmd/server/handler_test.go [hunk 5]
```diff
@@ -39,15 +39,15 @@ func TestMiddleware(t *testing.T) {
...
 	os.Setenv("NEW_RELIC_LICENSE_KEY", "1")
 	os.Setenv("NEW_RELIC_APP_NAME", "1")
 	var buffer bytes.Buffer
 	logrus.SetOutput(&buffer)
 	n := negroni.New()
 
 	useNewRelicMiddleware(n, false)
```
Comment (johnwu96822) [inline cmd/server/handler_test.go]: Need to take out the verbose param

**Comment by johnwu96822** [comment]
👍

**Comment by gubenyas** [comment]
[lockit v3.1.1] Merged via Lockit CLI on 2017-03-23 11:22:02.53648348 +0200 EET

**Pull request merged**